ygen/protogen.go

```diff
@@ -88,21 +101,69 @@ message {{ .Name }} {
 	}
 )
 
+// writeProtoHeader outputs the header for a proto3 generated file. The package
+// name, base for the generated packages, files for which the protobuf was
+// generated, and the paths used for include are supplied as arguments. In
+// addition, whether path compression is being performed, and the calling
+// binary name are provided as arguments. Returns a string containing the
```

Comment:
For such a long arg list it may be better split it into list format e.g. 
  pkg blah
  basePkg blah blah, set to defaultBasePackageName if empty
nit: go readabilty reviewers would insist on "It returns...". Up to you.

Reply:
Done.